### DISCUSSION TITLE
Xtext builds with Maven missing UI module

### DISCUSSION THREAD

**Discussion by farhan5248** [eclipse-xtext/xtext#2848] [Q&A] [answered]
Hi,

Not sure if this is the right place to ask this question but I figured I'd give it a try before trying reddit or stack overflow.
I followed the Xtext tutorials to create basic projects and everything is working fine.
When I try to create a project with the standard Maven structure (no separate test modules) it doesn't create the ui module, just the dsl one and the ide one.
Is that by design? I was looking for that module to put my syntax coloring classes etc.
Is there a sample project that you know of that demonstrates where all the classes go?

Thanks

**Comment by cdietrich** [comment]
Are you targeting eclipse or are you targeting language server or are you targeting the classic xtext-web

**Reply by farhan5248** [reply]
I'm targeting Eclipse only.

**Reply by cdietrich** [reply, marked as answer]
Eclipse uses eclipse plug-in project. There test projects are separate. Thus you should select the matching project layout
